### PR TITLE
added SHOW TABLES statement

### PR DESCRIPTION
**Pull request by celaus**

### CODE REVIEW

File: sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java [hunk 1]
```diff
@@ -634,9 +646,9 @@ private static void printStatement(String sql)
 
     private static void println(String s)
     {
-        if (Boolean.parseBoolean(System.getProperty("printParse"))) {
+       // if (Boolean.parseBoolean(System.getProperty("printParse"))) {
```
Comment (chaudum) [inline sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java]: ??

File: sql-parser/src/main/java/io/crate/sql/tree/ShowTables.java [hunk 2]
```diff
@@ -64,15 +78,24 @@ public boolean equals(Object obj)
...
+                .add("whereExpression", whereExpression.toString())
                 .toString();
     }
+
+    public Optional<String> likePattern() {
+        return likePattern;
+    }
+
+    public Optional<Expression> whereExpression() {
+        return this.whereExpression;
```
Comment (chaudum) [inline sql-parser/src/main/java/io/crate/sql/tree/ShowTables.java]: `this` not required

Comment (celaus) [inline sql-parser/src/main/java/io/crate/sql/tree/ShowTables.java]: why not?

File: sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java [hunk 3]
```diff
@@ -62,4 +64,47 @@ public AnalyzedStatement analyze(ShowSchemas node, Analysis analysis) {
...
+         * <code>
+         *     SELECT distinct(table_name) as table_name
+         *     FROM information_schema.tables
+         *     [ WHERE ( schema_name = 'schema_name' | schema_name NOT IN ( 'sys', 'information_schema' ) )
+         *      [ AND ( table_name LIKE 'pattern' | <where-clause > ) ]
+         *     ]
+         *     ORDER BY 1;
+         * </code>
+         */
+        StringBuilder sb = new StringBuilder("SELECT distinct(table_name) as table_name FROM information_schema.tables ");
```
Comment (mfelsche) [inline sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java]: consider creating an AstVisitor that creates String statements from Show... nodes.
Like this you can easily unittest this stuff and the parsing and select-analyzing can be refactored to a single method.

**Comment by mfelsche** [comment]
changes

**Comment by celaus** [comment]
Moved to #3074

**Pull request closed without merging**